### DISCUSSION TITLE
v0.3.9

### DISCUSSION THREAD

**Discussion by arxanas** [arxanas/git-branchless#261] [Release notes]
To install or update `git-branchless`, run the following:

```
$ cargo install --locked git-branchless
$ git branchless init  # in your repository
```

## Release highlights

- This release fixes a bug which several users reported. As always, thanks for your reports!

### Fixed

- (#249) On-disk `git move` commands now ensure that they keep your current branch checked out after the operation.
- (#249) On-disk `git move` commands no longer mysteriously move your current branch to a different commit in the stack.

<hr /><em>This discussion was created from the release <a href='https://github.com/arxanas/git-branchless/releases/tag/v0.3.9'>v0.3.9</a>.</em>

**Comment by thomcc** [comment]
Hmm, with `--locked` it gives me: 

```
warning: package `crossbeam-utils v0.8.5` in Cargo.lock is yanked in registry `crates-io`, consider running without --locked
```

Which is mildly concerning. Although, without `--locked` the build fails with

```
error[E0432]: unresolved import `cursive::traits::Boxable`
  --> /Users/thom/.cargo/registry/src/github.com-1ecc6299db9ec823/git-branchless-0.3.9/src/commands/undo.rs:14:5
   |
14 | use cursive::traits::Boxable;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^ no `Boxable` in `traits`
```

So I guess I'll be hoping that the yank was about some API not used here 😬 (it seems like it might have been a yank for semver reasons?)

**Reply by arxanas** [reply]
Thanks for raising this. Yeah, it looks like they automatically yank all older versions when they release a new version, so it shouldn't be a correctness/security issue in this case.

We use `--locked` because we depend on some prerelease versions of our dependencies, and Cargo treats all prerelease versions as compatible. I think all of our dependencies have exited prerelease, so it might be time to update on our end.

Unfortunately, it's not easy because of some breaking changes and some `tracing`-ecosystem dependencies which become runtime-incompatible.
